x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/RegressionIT.java

```diff
@@ -361,6 +361,9 @@ public void testStopAndRestart() throws Exception {
     public void testTwoJobsWithSameRandomizeSeedUseSameTrainingSet() throws Exception {
         String sourceIndex = "regression_two_jobs_with_same_randomize_seed_source";
         indexData(sourceIndex, 100, 0);
```

Comment:
`indexData` is calling directly `client().admin().indices().prepareCreate()`  instead of `prepareCreate()` of the test framework. This ensures that the index has always 1 shard.

However, it still can have multiple segments which then leads to non-deterministics order in which the reservoir sampling might get the documents. Hence, we need to fix both the shards and the segments to 1.